imsim/process_info.py

```diff
@@ -0,0 +1,54 @@
+import os
+import time
+from collections import namedtuple
+import psutil
+import galsim
+from galsim.config import ExtraOutputBuilder, RegisterExtraOutput
+
+
+_PROCESS_INFO_COLS = ['object_id', 'pid', 'rss', 'uss', 'user_time',
+                      'unix_time']
+_PROCESS_INFO_TYPES = [str, int, float, float, float, float]
+ProcessInfo = namedtuple('ProcessInfo', _PROCESS_INFO_COLS)
+
+
+class ProcessInfoBuilder(ExtraOutputBuilder):
+    """
+    Build output file with per-process cpu and memory info extracted
+    using psutil.
+    """
+    def processStamp(self, obj_num, config, base, logger):
+        object_id = base.get('object_id')
+
+        pid = os.getpid()
+        proc = psutil.Process(pid)
+        mem_full_info = proc.memory_full_info()
+
+        # Compute memory in GiB
+        rss = mem_full_info.rss/1024**3
+        uss = mem_full_info.uss/1024**3
+
+        # User cpu time
+        user_time = proc.cpu_times().user
+
+        process_info = ProcessInfo(object_id, pid, rss, uss, user_time,
+                                   time.time())
+        logger.info("Object %d, id %s, pid %d, RSS %.2f GB, USS %.2f GB, "
```

Comment:
So, if we set the log level to 2 it will write both to the log and the file correct?

If the level is set to 1 "warning" this won't be in the log but the file will still be created and written, is that correct?  Is there a way to tie these together so the extra output only shows up on 2?

Reply:
Thinking more about this, I can certainly imagine vases where we don’t want this coupled. But maybe in the YAML we could have a log level threshold for the the output files which is 2 by default.

Reply:
I don't understand the ask here.  If you don't want this as an extra_ouput, you just remove it from the config.  

Or do you want the option to have it in the config, so it makes logging statements, but not write a file?  If so, that's already possible by simply omitting the file_name from this extra_output.  Then it will run things, but not write to a file.

Reply:
Sorry.. I was unclear:

The option is now in the default config.  So, this means that users that don't understand they need to put a 

option: "" 

in their yaml files are always going to get this output (which can be hundreds of files).  So, I would like reasonable behavior for users while still being able to have the pattern etc setup in the the config file.  This is unlike checkpoints,  (which can always be usable If jobs need to restart), or centroids.  It's really a developer feature.

Ideally, it would be setup as now in the config file but the user would have to flip one option to have it start producing the output. I was suggestion perhaps tying it to a a minimum logging level so that by default they would start to show up when you asked for -v 2.

Reply:
I think we should not put it in the template config.  As developers, we know how to add it if we want it.

Reply:
> I think we should not put it in the template config. As developers, we know how to add it if we want it.

That's fine with me, but I think the format needs to be documented somehow.   Keeping it in `imsim-config.yaml`  but commented-out seems like a convenient way to do that and have it handy.

Reply:
Well, we have a documentation page.  It would be appropriate to add information about this type there.

Reply:
I added to the config.rst docs and put a full example there and deleted the entry from `imsim-config.yaml`

Reply:
Great. Thanks.

Another thing we could do if you wanted something particularly easy to copy/paste start from would be to add a new

imsim-developer.yaml

to the examples directory that is set up for doing tests etc.